tests/app/clients/test_freshdesk.py

```diff
@@ -0,0 +1,41 @@
+import base64
+
+import requests_mock
+
+from app.clients.freshdesk import Freshdesk
+
+
+def test_create_ticket(notify_api):
+    def match_json(request):
+        expected = {
+            'product_id': 42,
+            'subject': 'Ask a question',
+            'description': 'my message',
+            'email': 'test@example.com',
+            'priority': 1,
+            'status': 2,
+            'tags': []
+        }
+
+        encoded_auth = base64.b64encode(b'freshdesk-api-key:x').decode('ascii')
+        json_matches = request.json() == expected
+        basic_auth_header = request.headers.get('Authorization') == f"Basic {encoded_auth}"
```

Comment:
Everything looks good! I'm just curious how these two lines get their values. More specifically, how the `request` object has the json and header values assigned to it.

Reply:
This is some requests mock magic.

The line `additional_matcher=match_json` says that we should use the `match_json` method to check if the request that was sent was indeed the one we expected.

`Freshdesk.create_ticket` sends a request and `match_json` gets the sent request and analyzes it to see it was well-formatted (in our case auth headers + JSON body)